pkg/resource/sensitive.go

```diff
@@ -181,27 +202,58 @@ func GetSensitiveParameters(ctx context.Context, client SecretClient, from runti
 			if v == nil {
 				continue
 			}
-			sel := &v1.SecretKeySelector{}
-			if err = pavedJSON.GetValueInto(expandedJSONPath, sel); err != nil {
-				return errors.Wrapf(err, "cannot get SecretKeySelector from xp resource for fieldpath %q", expandedJSONPath)
-			}
-			sensitive, err = client.GetSecretValue(ctx, *sel)
-			if err != nil {
-				return errors.Wrapf(err, "cannot get secret value for %v", sel)
-			}
-			expTF, err := expandedTFPath(expandedJSONPath, mapping)
-			if err != nil {
-				return err
-			}
-			if err = pavedTF.SetString(expTF, string(sensitive)); err != nil {
-				return errors.Wrapf(err, "cannot set string as terraform attribute for fieldpath %q", tfPath)
+
+			if isSlice(v) {
+				sel := &[]v1.SecretKeySelector{}
+				if err = pavedJSON.GetValueInto(expandedJSONPath, sel); err != nil {
+					return errors.Wrapf(err, "cannot get SecretKeySelector from xp resource for fieldpath %q", expandedJSONPath)
+				}
+				var sensitives []interface{}
+				for _, s := range *sel {
+					sensitive, err = client.GetSecretValue(ctx, s)
+					if err != nil {
+						return errors.Wrapf(err, "cannot get secret value for %v", sel)
+					}
+					sensitives = append(sensitives, string(sensitive))
+				}
+				expTF, err := expandedTFPath(expandedJSONPath, mapping)
+				if err != nil {
+					return err
+				}
+				if err = pavedTF.SetValue(expTF, sensitives); err != nil {
+					return errors.Wrapf(err, "cannot set string as terraform attribute for fieldpath %q", tfPath)
+				}
+			} else {
+				sel := &v1.SecretKeySelector{}
+				if err = pavedJSON.GetValueInto(expandedJSONPath, sel); err != nil {
+					return errors.Wrapf(err, "cannot get SecretKeySelector from xp resource for fieldpath %q", expandedJSONPath)
+				}
+				sensitive, err = client.GetSecretValue(ctx, *sel)
+				if err != nil {
+					return errors.Wrapf(err, "cannot get secret value for %v", sel)
+				}
+				expTF, err := expandedTFPath(expandedJSONPath, mapping)
+				if err != nil {
+					return err
+				}
+				if err = pavedTF.SetString(expTF, string(sensitive)); err != nil {
+					return errors.Wrapf(err, "cannot set string as terraform attribute for fieldpath %q", tfPath)
+				}
 			}
 		}
 	}
 
 	return nil
 }
 
+func isSlice(v interface{}) bool {
```

Comment:
We may replace this function with a type switch as you suggested.

Reply:
Done! Using type switch will provide easy maintenance for new sensitive field types we plan to add in the future.